compat/test_suite.c

```diff
@@ -0,0 +1,94 @@
+/* vi:set ts=4 sw=4 expandtab:
+*
+* Copyright 2016, Chris Leishman (http://github.com/cleishm)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "../config.h"
+#include "../lib/src/cypher-parser.h"
+#include <check.h>
+#include <errno.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+TCase* call_tcase(void);
```

Comment:
This seems to duplicate the autogenerated `lib/src/check_libcypher-parser_suite.c`? Can we avoid that duplication and avoid checking in an autogenerated file?

Reply:
Yes, I see two solutions.  Using a common template file and replace variables from both Makefile.am and CMakeLists.txt, or porting the check_libcypher-parser_suite.c rule to CMake.

I experimented with the first option. Which one would you prefer?

C Template File
```
#include <check.h>

@cases@

Suite *libcypher_parser_suite(void)
{
	Suite *s = suite_create("libcypher-parser");
	@add_cases@
	return s;
}
```

Replace template variables in Makefile.am
```
check_libcypher-parser_suite.c: ${check_libcypher_parser_CHECKS}
	@(for check in $^; do \
		name=$$(basename $$check .c | sed -e 's/^check_//'); \
		cases="$${cases}TCase* $${name}_tcase(void);@"; \
		add_cases="$${add_cases}suite_add_tcase(s, $${name}_tcase());:"; \
	done; \
	cases=$$(echo $${cases} | sed -e 's/@$$//'); \
	add_cases=$$(echo $${add_cases} | sed -e 's/:$$//'); \
	nlinit=`echo 'nl="'; echo '"'`; eval "$$nlinit"; \
	sed check_libcypher-parser_suite.c.in -e 's/@cases@/'"$${cases}"'/g' -e 's/@add_cases@/'"$${add_cases}"'/g' | \
	sed -e "s/@/\\$${nl}/g" | sed -e "s/:/\\$${nl}    /g"  ) > $@
```

CMake code to come.

Reply:
Generating the file from either/both build systems seems to make sense.